sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -78,6 +78,27 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     }
   }
 
+  private def toSQL(node: LogicalPlan, topNode: Boolean): String = {
```

Comment:
Yea, I see. Join is the top node only when using DataFrame API. But the original purpose of this change is not intended to support converting logical plan of DataFrame to SQL. The test case is just used to construct the Join plan needed to test it. This change is introduced to support the option to convert a logical plan not beginning from the top of the whole plan.